api/src/main/java/org/apache/iceberg/expressions/Literals.java

```diff
@@ -126,7 +127,17 @@ public final ByteBuffer toByteBuffer() {
 
     @Override
     public String toString() {
-      return String.valueOf(value);
+      if (value == null) {
+        return "null";
+      }
+
+      switch (typeId()) {
+        case FIXED:
+        case BINARY:
+          return "0x" + BaseEncoding.base16().encode((toByteBuffer().array()));
+        default:
+          return String.valueOf(value);
+      }
```

Comment:
This method should not change. It is the base implementation. Instead, you should override the implementation in `BinaryLiteral` and `FixedLiteral`.

Reply:
ok, thanks for your reply